File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
object ScrabbleScore {

    fun scoreLetter(c: Char): Int {
        return when(c.toUpperCase()) {
            in "AEIOULNRST" -> 1
            in "DG"         -> 2
            in "BCMP"       -> 3
            in "FHVWY"      -> 4
            in "K"          -> 5
            in "JX"         -> 8
            in "QZ"         -> 10
            else            -> 0
        }
    }

    fun scoreWord(word: String): Int = word.sumBy { scoreLetter(it) }
}

//        return word.fold(0) { score, c -> score + scoreLetter(c) }

File: fsharp/scrabble-score/ScrabbleScore.fs
module ScrabbleScore

let score (word: string): int = 
    let letterScore =
        function
        | 'A' | 'E' | 'I' | 'O' | 'U' -> 1
        | 'L' | 'N' | 'R' | 'S' | 'T' -> 1
        | 'D' | 'G' -> 2
        | 'B' | 'C' | 'M' | 'P' -> 3
        | 'F' | 'H' | 'V' | 'W' | 'Y' -> 4
        | 'K' -> 5
        | 'J' | 'X' -> 8
        | 'Q' | 'Z' -> 10
        | _ -> 0

    word.ToUpper() 
    |> Seq.sumBy letterScore
